app/app/home.tsx

```diff
@@ -33,50 +33,115 @@ const PRICE_LABEL = {
   ),
 };
 
+type HomeState = {
+  zone: BiddingZone | null;
+  is_fetching_price: boolean;
+  price: number | null;
+  error: Error | null;
+};
+
+export interface HomeController {
+  state: HomeState;
+  loadBiddingZone: (zone: BiddingZone) => void;
+}
+
 export default function Home() {
-  const [selectedZone, setSelectedZone] = useState<Status>(statuses[0]);
-
-  const { isPending, error, data, refetch } = useQuery({
-    queryKey: ['currentPrice'],
-    queryFn: async () => {
-      const currTime = new Date();
-      const year = currTime.getFullYear();
-      const month = currTime.getMonth() + 1;
-      const day = currTime.getDate();
-      const hour = currTime.getHours();
-
-      const URL =
-        'https://www.elprisetjustnu.se/api/v1/prices/' +
-        year +
-        '/0' +
-        month +
-        '-' +
-        day +
-        '_' +
-        selectedZone.value +
-        '.json';
-
-      const response = await fetch(URL);
-
-      const data = await response.json();
-
-      return data[hour]['SEK_per_kWh'] as number;
-    },
+  const [homeState, setHomeState] = useState<HomeState>({
+    zone: null,
+    is_fetching_price: false,
+    price: null,
+    error: null,
   });
 
-  useEffect(() => {
-    refetch();
+  const regionSelectControllerRef = useRef<RegionSelectController>(null);
+
+  // Call external api to get price for zone
+  const price_api_call = async (zone: BiddingZone) => {
+    function delay(ms: number) {
+      return new Promise((resolve) => setTimeout(resolve, ms));
+    }
+
+    await delay(500);
+
+    const currTime = new Date();
+    const year = currTime.getFullYear();
+    const month = currTime.getMonth() + 1;
+    const day = currTime.getDate();
+
+    const URL =
+      'https://www.elprisetjustnu.se/api/v1/prices/' + year + '/0' + month + '-' + day + '_' + zone.value + '.json';
+
+    return fetch(URL);
+  };
+
+  const home_controller = useRef<HomeController>({
+    state: homeState,
+    loadBiddingZone: async (zone) => {
+      home_controller.current.state = {
+        ...home_controller.current.state,
+        zone: zone,
+        price: null,
+        is_fetching_price: true,
+        error: null,
+      };
+
+      // Price starts loading, update state
+      regionSelectControllerRef.current?.setRegionLoaded(false);
+      setHomeState(home_controller.current.state);
+
+      const response = await price_api_call(zone);
+
```

Comment:
Maby use try catch for error handeling like timeouts, as you are currently only handling ok responses.